src/main/java/org/swisspush/redisques/RedisQues.java

```diff
@@ -919,9 +953,11 @@ private void readQueue(final String queue) {
                             } else {
                                 // Failed. Message will be kept in queue and retried later
                                 log.debug("RedisQues Processing failed for queue " + queue);
-                                myQueues.put(queue, QueueState.READY);
```

Comment:
Just wondering: Why do you now delay the `QueueState.READY` until the retry timer fires? (I see that you do it now in the method `rescheduleSendMessageAfterFailure(...)`)
I must say that I don't understand the utility value of having two Queue-States. I mainly see: Is there a consumer for this queue or not (yet).
Thanks for some clarifying words.

Reply:
How I understand it: If we would put the state to READY at this stage then the queue would be picked up by the next cycle of a periodic refresh which we don't want, so we keep it in state CONSUMING until the "retry timer" expires.